tests/gis_tests/test_utils.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from django.db import connection, models
+from django.db.models.expressions import Func
+from django.test import SimpleTestCase
+
+from .utils import FuncTestMixin
+
+
+def test_mutation(mutation_func):
+    vendor_impl = 'as_' + connection.vendor
+
+    def wrapper(*args, **kwargs):
+        class TestFunc(Func):
+            output_field = models.IntegerField()
+
+            def __init__(self):
+                self.attribute = 'initial'
+                super().__init__('initial', ['initial'])
+
+            def as_sql(self, *args, **kwargs):
+                mutation_func(self)
+                return '1', ()
+
+        getattr(TestFunc(), vendor_impl)(None, None)
+
+    return wrapper
+
+
+class FuncTestMixinTest(FuncTestMixin, SimpleTestCase):
+    @unittest.expectedFailure
```

Comment:
I had some difficulty understanding how this test works, but is it possible to use an approach that would allow using  `assertRaisesMessage(AssertionError, '...')`  instead of `expectedFailure`? I've had [some doubts](https://code.djangoproject.com/ticket/23943) in the past that `expectedFailure` is the best approach since things can fail in many ways besides the failure you expect.